frontend/ui/widget/virtualkeyboard.lua

```diff
@@ -32,14 +32,14 @@ local VirtualKey = InputContainer:new{
 }
 
 function VirtualKey:init()
-    if self.label == "Sym" or self.label == "ABC" then
-        self.callback = function () self.keyboard:setLayout(self.key or self.label) end
+    if self.label == "Sym" or self.label == "ABC" or self.label == "かな" or self.label == "記号" then
```

Comment:
The design looks good to me :) @alexandrahurst since you are already at it, do you mind doing the refactoring too? It would make the design much more flexible to support other languages going forward.